### PR TITLE
Hydra yaml config

### PR DESCRIPTION
**Pull request by kks32**
Use yaml to configure settings rather than command line arguments:

- [x] Define input file schema and validation
- [x] Support resume flag
- [x] Update README
- [x] move constants to data section

**Related Issues/PRs**
#80

### CODE REVIEW

File: gns/train.py [hunk 1]
```diff
@@ -165,46 +105,46 @@ def rollout(
...
+        os.makedirs(cfg.output.path)
 
     # Use `valid`` set for eval mode if not use `test`
     split = (
         "test"
-        if (
-            FLAGS.mode == "rollout"
-            or (not os.path.isfile("{FLAGS.data_path}valid.npz"))
-        )
+        if (cfg.mode == "rollout" or (not os.path.isfile("{cfg.data_path}valid.npz")))
```
Comment (yjchoi1) [inline gns/train.py:142]: `"{cfg.data_path}valid.npz"` to `f"{cfg.data_path}valid.npz"`

**Comment by yjchoi1** [comment]
Would it be better to add the configurations from metadata.json to config.yml? However, this might cause compatibility issues with the previous version, which explicitly reads from `metadata.json`.

Currently, `metadata.json` controls some of the important hyperparameters like `n_message_passing`, `connectivity_raidus`, `nnode_in` (optional) etc.

File: gns/train.py (hunk 1 above)
Comment (kks32) [reply gns/train.py]: I assume you mean `{cfg.data.path}valid.npz`

Comment (yjchoi1) [reply gns/train.py]: Shouldn't it be `f"{cfg.data_path}valid.npz"` instead of `"{cfg.data_path}valid.npz"`?

**Pull request merged**